### ISSUE TITLE
Support of key bindings/hotkeys

### ISSUE THREAD

**Issue by christianguenter2** [larshp/abapGit#1728]
To improve developer productivity and for a more seamless keyboard navigation it would be good to support key bindings. It should be possible to maintain them via settings dialog.

A few examples how the config could look like:

key sequence | action
-----------|--------------------
o | overview page
CTRL+r | refresh repo
CTLR+SHIFT+s | settings

**Comment by christianguenter2** [comment]
[WIP](https://github.com/christianguenter2/abapGit/tree/exp_keybindings)

![image](https://user-images.githubusercontent.com/17437789/43790072-debcd57a-9a71-11e8-8cd6-6f24b063c6b6.png)

![image](https://user-images.githubusercontent.com/17437789/43790149-0901ef14-9a72-11e8-8778-61733454b8e5.png)

**Comment by larshp** [comment]
how does it know the actions?

**Comment by christianguenter2** [comment]
Currently they are dynamically extracted from ZIF_ABAPGIT_DEFINITIONS=>GC_ACTION and ZCL_ABAPGIT_GUI_PAGE_MAIN=>C_ACTIONS

https://github.com/christianguenter2/abapGit/blob/4811493d66c6dcaf66c7cd5e9334ee1b31c83168/src/ui/zcl_abapgit_gui_page_settings.clas.abap#L571-L614

**Comment by larshp** [comment]
I'd like to reduce the size of the router, which means reducing GC_ACTION, as it is a place that will cause conflicts, and the actions are typically dependent on how the pages are made.

Guess the above only works for global classes?

**Comment by christianguenter2** [comment]
It also work for local classes/interfaces. Agree with reducing the size of the router and GC_ACTION, I think we could easily add other structured constants to my implementation. It's isn't set in stone. Or do you think we should explicitly specify which actions are available for hotkeys?

**Comment by larshp** [comment]
Suggest adding a static method to ZIF_ABAPGIT_GUI_PAGE under which the page can register default hotkeys/actions, however this requires being able to determine which pages exist

**Comment by christianguenter2** [comment]
Good idea, I try to follow this path.

> however this requires being able to determine which pages exist

All pages are subclasses of zcl_abapgit_gui_page?

**Comment by larshp** [comment]
subclasses: yes

but I do not know of any way to find subclasses for local classes

**Comment by christianguenter2** [comment]
hm agree, that might be a problem

**Comment by larshp** [comment]
there might be a way to do it, also not having the hotkeys feature in the compiled might be okay

**Comment by christianguenter2** [comment]
![image](https://user-images.githubusercontent.com/17437789/44053687-b36f8934-9f40-11e8-8df0-c59c06c06986.png)
